src/calendar/MonthPicker.ts

```diff
@@ -0,0 +1,299 @@
+import { WidgetBase } from '@dojo/widget-core/WidgetBase';
+import { ThemeableMixin, ThemeableProperties, theme } from '@dojo/widget-core/mixins/Themeable';
+import { v, w } from '@dojo/widget-core/d';
+import uuid from '@dojo/core/uuid';
+import { Keys } from '../common/util';
+import { DNode, TypedTargetEvent } from '@dojo/widget-core/interfaces';
+import Radio from '../radio/Radio';
+import Button from '../button/Button';
+import * as css from './styles/calendar.m.css';
+import * as baseCss from '../common/styles/base.m.css';
+
+// will need to localize messages
+export interface CalendarMessages {
+	chooseMonth: string;
+	chooseYear: string;
+	previousMonth: string;
+	nextMonth: string;
+};
+
+/**
+ * @type MonthPickerProperties
+ *
+ * Properties that can be set on a Calendar component
+ *
+ * @property labelId              Set id to reference label containing current month and year
+ * @property labels               Customize or internationalize accessible helper text
+ * @property month                Currently displayed month, zero-based
+ * @property monthNames           Array of full and abbreviated month names
+ * @property open                 Boolean, sets state of popup
+ * @property renderMonthLabel     Format the displayed current month and year
+ * @property year                 Currently displayed year
+ * @property onRequestOpen        Called when a user action occurs that should trigger the popup opening
+ * @property onRequestClose       Called when a user action occurs that should close the popup
+ * @property onRequestMonthChange Called when a month should change; receives the zero-based month number
+ * @property onRequestYearChange  Called when a year should change; receives the year as an integer
+ */
+export interface MonthPickerProperties extends ThemeableProperties {
+	labelId?: string;
+	labels: CalendarMessages;
+	month: number;
+	monthNames: { short: string; long: string; }[];
+	open?: boolean;
+	renderMonthLabel?(month: number, year: number): string;
+	year: number;
+	onRequestOpen?(): void;
+	onRequestClose?(): void;
+	onRequestMonthChange?(month: number): void;
+	onRequestYearChange?(year: number): void;
+};
+
+export const MonthPickerBase = ThemeableMixin(WidgetBase);
+
+@theme(css)
+export default class MonthPicker extends MonthPickerBase<MonthPickerProperties> {
```

Comment:
Hm, I guess I didn't really see it as something that people would customize. Right now everything in there apart from `render` is private, so it's not easily extendable. I wonder if I should start making more things protected over private, both in MonthPicker and in general?